File: super.py
# super.py

class Super:
    '''
    Minimal, one-line implementations of algorithms solving some basic mathematical problems.
    '''
    power = staticmethod(lambda n, e: 1 if e == 0 else n if e == 1 else n * Super.power(n, e - 1))
    digital_root = staticmethod(lambda n: n if n < 10 else Super.digital_root(sum(map(int, str(n)))))
    dec2bin = staticmethod(lambda d: '0' if d == 0 else Super.dec2bin(d // 2) + str(d % 2))
    bin2dec = staticmethod(lambda b: 0 if not b else Super.power(2, len(b) - 1) * int(b[0]) + Super.bin2dec(b[1:]))
    dec2hex = staticmethod(lambda d: '0' if d == 0 else Super.dec2hex(d // 16) + chr(d % 16 + (ord('0') if d % 16 < 10 else ord('A') - 10)))
    hex2dec = staticmethod(lambda h: 0 if not h else (ord(h[0]) - (ord('0') if ord(h[0]) < ord('A') else ord('A') - 10)) * Super.power(16, len(h) - 1) + Super.hex2dec(h[1:]))
    prime_check = staticmethod(lambda n, _cur=2: False if n < 2 else True if _cur == n else False if n % _cur == 0 else Super.prime_check(n, _cur + 1))
    @staticmethod
    def primes_list(max: int, _prms: [int] = [2], _cur: int = 3) -> [int]:
        return _prms if _cur > max else Super.primes_list(max, (_prms if next((p for p in _prms if _cur % p == 0), None) else _prms + [_cur]), _cur + 2)

File: super.php
<?php
/**
 * Class Super
 *
 * Minimal, one-line implementations of algorithms solving some basic mathematical problems.
 */
class Super {
    static function power($n,$e)        { return $e==0 ? 1 : $e==1 ? $n : $n*self::power($n,$e-1); }
    static function digital_root($n)    { return $n<10 ? $n : self::digital_root(array_sum(str_split($n))); }
    static function dec2bin($d)         { return $d==0 ? "0" : self::dec2bin(intval($d/2)) . ($d%2); }
    static function bin2dec($b)         { return $b=="" ? 0 : (intval($b[0]))*pow(2, strlen($b)-1) + self::bin2dec(substr($b,1)); }
    static function dec2hex($d)         { return $d==0 ? "0" : self::dec2hex(intval($d/16)) . ($d%16<10 ? chr(48+($d%16)) : chr(65+($d%16-10))); }
    static function hex2dec($h)         { return $h=="" ? 0 : ((ord($h[0])<65 ? ord($h[0])-48 : ord($h[0])-55)*pow(16, strlen($h)-1)) + self::hex2dec(substr($h,1)); }
    static function is_prime($n,$cur=2) { return $n<2 ? false : $n==$cur ? true : $n%$cur==0 ? false : self::is_prime($n,$cur+1); }
	static function primes_list($max,$prms=[2],$cur=3) {
		return $cur>$max ? $prms : self::primes_list($max, (!array_filter($prms, fn($p)=>$cur%$p==0) ? [...$prms,$cur] : $prms), $cur+2);
	}
}
?>
